org.eclipse.dawnsci.analysis.dataset/src/org/eclipse/dawnsci/analysis/dataset/SliceStreamSupport.java

```diff
@@ -0,0 +1,142 @@
+ * 
+ * <pre>
+ * <code>
+final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+List<Number> maxes = SliceStreamSupport.sliceStream(lz, 2, 3).map(set->set.max()).collect(Collectors.toList());
+  </code>
+ *</pre> 
+ * @see StreamSupport 
+ */
+public class SliceStreamSupport {
+
+	private static final int FLAGS = Spliterator.SIZED | Spliterator.DISTINCT | Spliterator.CONCURRENT
+			| Spliterator.IMMUTABLE | Spliterator.NONNULL;
+
+	/**
+	 * Create a dataset stream from a Lazy Dataset by specifying the axes that you
+	 * would like to size over.
+	 * 
+	 * @param lazy
+	 * @param omit
+	 * @return a stream of lazy data
+	 */
+	public static Stream<ILazyDataset> stream(ILazyDataset lazy, int... omit) {
+		return stream(lazy, false, omit);
+	}
+	
+	/**
+	 * 
+	 * @param lazy
+	 * @param omit
+	 * @param parallel
+	 * @return stream of lazy datasets
+	 */
+	public static Stream<ILazyDataset> stream(ILazyDataset lazy, boolean parallel, int... omit) {
+		SliceViewIterator it = new SliceViewIterator(lazy, null, omit);
+		return StreamSupport.stream(new LazyDatasetSpliterator(it), parallel);
+	}
+
+	/**
+	 * Create a dataset stream from a Lazy Dataset by specifying the axes that you
+	 * would like to size over.
+	 * 
+	 * @param lazy
+	 * @param omit
+	 * @return a stream of lazy data
+	 */
+	public static Stream<IDataset> sliceStream(ILazyDataset lazy, int... omit) {
+		return sliceStream(lazy, false, omit);
+	}
+	
+	/**
+	 * 
+	 * @param lazy
+	 * @param omit
+	 * @param parallel
+	 * @return stream of lazy datasets
+	 */
+	public static Stream<IDataset> sliceStream(ILazyDataset lazy, boolean parallel, int... omit) {
+		SliceViewIterator it = new SliceViewIterator(lazy, null, omit);
+		return StreamSupport.stream(new DatasetSpliterator(it), parallel);
+	}
+
+
+	private static class LazyDatasetSpliterator extends Spliterators.AbstractSpliterator<ILazyDataset> {
+
+		private final SliceViewIterator it;
+
+		public LazyDatasetSpliterator(SliceViewIterator it){
+			super(it.getTotal(),FLAGS);
+			this.it = it;
+		}
+
+		@Override
+		public boolean tryAdvance(Consumer<? super ILazyDataset> action) {
+			if (action == null) {
+				throw new NullPointerException();
+			}
+
+			boolean hasNext = it.hasNext();
+			if (hasNext) {
+				action.accept(it.next());
+			}
+
+			return hasNext;
+		}
+	}
+
+	private static class DatasetSpliterator extends Spliterators.AbstractSpliterator<IDataset> {
+
+		private final SliceViewIterator it;
+
+		public DatasetSpliterator(SliceViewIterator it){
+			super(it.getTotal(),FLAGS);
+			this.it = it;
+		}
+
+		@Override
+		public boolean tryAdvance(Consumer<? super IDataset> action) {
+			if (action == null) {
+				throw new NullPointerException();
+			}
+
+			boolean hasNext = it.hasNext();
+			if (hasNext) {
+				try {
+					action.accept(it.next().getSlice());
+				} catch (DatasetException e) {
+					throw new RuntimeException("Could not slice dataset",e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)